Pipeline Refactor - Take 2 Part 2

ranj063 opened the pull request.

ranj063 committed:
ASoC: SOF: sof-audio: Add helper functions for BE managed pipelines

Define 3 new helper functions for setting up/freeing/unpreparing BE
pipelines. These will be used when the pipeline management is
refactored to set up/free widgets, routes and pipelines during BE DAI
prepare or hw_free.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

ranj063 committed:
ASoC: SOF: Refactor pipeline management

Invoke the BE pipeline helper functions in the BE DAI prepare and hw_free
to ensure that all the widgets and routes belonging to the BE pipeline are
handled in the BE DAI ops. Add a new field in struct snd_sof_pipeline to
identify BE managed pipelines i.e. those that will managed by the BE DAI
ops. Lastly, modify the FE ops to make sure that the widgets/routes
belonging to these pipelines are skipped during setup/prepare/unprepare.
Note that the widget_ipc_prepare op handles all pipelines including
those managed by the BE DAI ops. This is because in order to prepare all
the widgets, the hw_params needs to be propagated along the list of
connected DAPM widgets in order from the source to the sink.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

ranj063 committed:
ASoC: SOF: ipc4-topology: Remove the skip_during_fe_trigger flag

Now that all widgets/routes/pipelines are setup, triggered and freed in
the BE DAI ops, we can reuse the be_pipeline flag in struct
snd_sof_pipeline to skip triggering the BE pipelines in the PCM trigger.
So remove the skip_during_fe_trigger flag from struct sof_ipc4_pipeline.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

ranj063 committed:
drivers: soundwire: Add new prepare op in struct sdw_intel_ops

Add and define the prepare op for Intel ACE2.x platforms. For the
moment, the prepare_stream op is exactly the same as the params_stream
op. But it will be modified in the following patch when the pipeline
management is refactored.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

lgirdwood reviewed (commented):
@ranj063 can you ask validation to schedule a stress test on this PR using latest stable FW release. Thanks

ranj063 commented:
SOFCI TEST

ranj063 committed:
ASoC: SOF: ipc4-topology: reset pipeline mem_usage during unprepare

The mem_usage for a pipeline is updated when the widgets in the
pipeline are prepare. So, reset it when a widget in the pipeline is
unprepared. The ipc_unprepare for the pipeline widget might be called
multiple times when each widget associated with the pipeline is
unprepared and it is harmless.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

ranj063 committed:
Fix for IPC timeouts
squash me with earlier commits
Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
